library/src/app/components/transfer/transfer-confirm/transfer-confirm.component.ts

```diff
@@ -0,0 +1,190 @@
+  error$: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
+
+  unsubscribe$ = new Subject();
+
+  pollConfig: PollConfig = {
+    timeout: this.error$,
+    interval: Constants.POLL_INTERVAL,
+    duration: Constants.POLL_DURATION
+  };
+
+  constructor(
+    @Inject(MAT_DIALOG_DATA)
+    public data: TransferConfirmData,
+    public dialogRef: MatDialogRef<TransferConfirmComponent>,
+    private snackBar: MatSnackBar,
+    private configService: ConfigService,
+    private quoteService: QuotesService,
+    private transfersService: TransfersService,
+    private eventService: EventService,
+    private errorService: ErrorService,
+    private translatePipe: TranslatePipe
+  ) {}
+
+  ngOnInit() {
+    this.getQuote();
+  }
+
+  getQuote(): void {
+    this.eventService.handleEvent(
+      LEVEL.INFO,
+      CODE.DATA_FETCHING,
+      'Fetching quote...'
+    );
+    interval(Constants.PLATFORM_REFRESH_INTERVAL)
+      .pipe(
+        takeUntil(this.unsubscribe$),
+        switchMap(() =>
+          this.quoteService.createQuote(this.data.postQuoteBankModel)
+        ),
+        map((quote) => this.quote$.next(quote)),
+        catchError((err) => {
+          this.dialogRef.close();
+          const message = this.translatePipe.transform(
+            'trade.confirm.error.quote'
+          );
+          this.snackBar.open(message, 'OK');
+          this.eventService.handleEvent(
+            LEVEL.ERROR,
+            CODE.DATA_ERROR,
+            'Error fetching quote'
+          );
+          this.errorService.handleError(err);
+          return of(err);
+        })
+      )
+      .subscribe();
+  }
+
+  onConfirmTransfer(): void {
+    this.isLoading$.next(true);
+
+    const postTransferBankModel: PostTransferBankModel = {
+      external_bank_account_guid: this.data.externalBankAccountBankModel.guid,
+      quote_guid: this.quote$.getValue().guid!,
+      transfer_type: PostTransferBankModel.TransferTypeEnum.Funding
+    };
+
+    let poll: Poll = new Poll(this.pollConfig);
+    let transfer: TransferBankModel;
+
+    this.error$.pipe(takeUntil(this.unsubscribe$)).subscribe((res) => {
+      if (res) {
+        this.dialogRef.close();
+        this.snackBar.open(
+          this.translatePipe.transform('transfer.error'),
+          'OK'
+        );
+        this.eventService.handleEvent(
+          LEVEL.ERROR,
+          CODE.DATA_ERROR,
+          'Error creating transfer'
+        );
+        this.errorService.handleError(new Error('Error creating transfer'));
+      }
+    });
+
+    this.transfersService
+      .createTransfer(postTransferBankModel)
+      .pipe(
+        switchMap((res) => {
+          transfer = res;
+          return poll.start();
+        }),
+        switchMap(() => this.transfersService.getTransfer(transfer.guid!)),
+        takeUntil(merge(poll.session$, this.unsubscribe$)),
+        skipWhile((transfer) => transfer.state === 'storing'),
+        map((transfer) => {
+          poll.stop();
```

Comment:
The transfer is going to come back as either `failed` or `pending` during this window. In sandbox you'll also see it come back as `completed`, but that's not going to happen in production for a long period of time.

We should handle all cases of course, but we should notify the user that a transfer is `pending` and that they'll see a pending balance until it is posted. At which point the balance will become `available`.

Reply:
I've made some changes here. After a transfer is created, the state is displayed in the details dialog. For the states: `storing`, `initiating`, and `pending` the ui now shows `Processing`. The states `failed` and `completed` are displayed individually.